include/engine/routing_algorithms.hpp

```diff
@@ -184,6 +184,32 @@ inline routing_algorithms::SubMatchingList RoutingAlgorithms<Algorithm>::MapMatc
 }
 
 template <typename Algorithm>
+std::vector<EdgeDuration> RoutingAlgorithms<Algorithm>::ManyToManySearch(
+    const std::vector<PhantomNode> &phantom_nodes,
+    const std::vector<std::size_t> &_source_indices,
```

Comment:
as we are copying anyhow, whouldn't it make sense to take by value? This way we could move from the outside, if we don't want to keep the data, also if we don't want to specify them, it would be clearer not to pass in an empty array

Reply:
just looked at `RoutingAlgorithmsInterface` and looks like it is a common pattern to pass const references in interfaces. I will keep the original interface and copy vectors explicitly instead of hiding copy assignment in arguments code.